pkg/api/meta/help.go

```diff
@@ -117,6 +125,13 @@ func SetList(list runtime.Object, objects []runtime.Object) error {
 	slice := reflect.MakeSlice(items.Type(), len(objects), len(objects))
 	for i := range objects {
 		dest := slice.Index(i)
+
+		// check to see if you're directly assignable
+		if reflect.TypeOf(objects[i]).AssignableTo(dest.Type()) {
```

Comment:
When you pass an `UnstructuredList` into this method, you don't want to dereference the items like you do in `EnforcePtr` below.